src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,39 +1,70 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+
+type Props = {
+  selectedTodo: Todo;
+  setSelectedTodo: (todo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [loader, setLoader] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    getUser(selectedTodo.userId).then(fetchedUser => {
+      setUser(fetchedUser);
+      setLoader(false);
+    });
+  });
```

Comment:
You're not providing a dependency array to `useEffect`. Without it, the effect will run after every render, which can lead to performance issues. Consider passing `[selectedTodo.userId]` as a dependency array to ensure that the effect runs only when `selectedTodo.userId` changes.